src/AlgebraicGeometry/ToricVarieties/ToricSchemes/attributes.jl

```diff
@@ -0,0 +1,136 @@
+    n = length(hilbert_basis(C))
+    R, _ = polynomial_ring(QQ, ["x_$(i)_$(k)" for i in 1:n], cached = false);
+    set_attribute!(A, :toric_ideal, toric_ideal(R, A))
+  end
+  cov = Covering(patch_list)
+  
+  for i in 1:(length(patch_list)-1)
+    for j in i+1:length(patch_list)
+      
+      X = patch_list[i]
+      Y = patch_list[j]
+      
+      # Step 1: Only glue if the cone of intersection has the "right" dimension:
+      facet = intersect(cone(X), cone(Y))
+      (dim(facet) == dim(cone(X)) - 1) || continue
+      
+      # Step 2: Find localization element vmat
+      CXdual = dual_cone(X)
+      CYdual = dual_cone(Y)
+      candidates = polarize(facet).pm_cone.HILBERT_BASIS_GENERATORS[2]
+      pos = findfirst(j -> ((candidates[j,:] in CXdual) && ((-candidates[j,:]) in CYdual)), 1:nrows(candidates))
+      sign = 1
+      if pos === nothing
+        pos = findfirst(j -> (((-candidates[j,:]) in CXdual) && (candidates[j,:] in CYdual)), 1:nrows(candidates))
+        sign = -1
+      end
+      @req pos !== nothing "no element found for localization"
+      vmat = sign * matrix(ZZ.(collect(candidates[pos,:])))
+      
+      # Step 3: Localize X at vmat
+      AX = transpose(hilbert_basis(X))
+      Id = identity_matrix(ZZ, ncols(AX))
+      sol = solve_mixed(AX, vmat, Id, zero(vmat))
+      fX = prod([x^k for (x, k) in zip(gens(ambient_coordinate_ring(X)), sol)])
+      U = PrincipalOpenSubset(X, OO(X)(fX))
+      
+      # Step 4: Localize Y at vmat
+      AY = transpose(hilbert_basis(Y))
+      Id = identity_matrix(ZZ, ncols(AY))
+      sol = solve_mixed(AY, -vmat, Id, zero(vmat))
+      fY = prod([x^(k>0 ? 1 : 0) for (x, k) in zip(gens(ambient_coordinate_ring(Y)), sol)])
+      V = PrincipalOpenSubset(Y, OO(Y)(fY))
+      
+      # Step 5: Compute the glueing isomorphisms V -> U
+      l = findfirst(j -> (vmat == AX[:,j]), 1:ncols(AX))
+      Idext = identity_matrix(ZZ, ncols(AX))
+      Idext[l,l] = 0
+      img_gens = [solve_mixed(AX, AY[:, k], Idext, zero(matrix_space(ZZ, ncols(Idext), 1))) for k in 1:ncols(AY)]
+      fres = hom(OO(V), OO(U), [prod([(k >= 0 ? x^k : inv(x)^(-k)) for (x, k) in zip(gens(OO(U)), w)]) for w in img_gens])
+      
+      # Step 6: Compute the glueing isomorphisms U -> V
+      l = findfirst(j -> (-vmat == AY[:,j]), 1:ncols(AY))
+      Idext = identity_matrix(ZZ, ncols(AY))
+      Idext[l,l] = 0
+      img_gens = [solve_mixed(AY, AX[:, k], Idext, zero(matrix_space(ZZ, ncols(Idext), 1))) for k in 1:ncols(AX)]
+      gres = hom(OO(U), OO(V), [prod([(k >= 0 ? x^k : inv(x)^(-k)) for (x, k) in zip(gens(OO(V)), w)]) for w in img_gens])
+      
+      # Step 7: Assign the gluing isomorphisms
+      set_attribute!(gres, :inverse, fres)
+      set_attribute!(fres, :inverse, gres)
+      f = SpecMor(U, V, fres)
+      g = SpecMor(V, U, gres)
+      set_attribute!(g, :inverse, f)
+      set_attribute!(f, :inverse, g)
+      G = SimpleGlueing(X, Y, f, g)
+      add_glueing!(cov, G)
```

Comment:
Why one giant for loop? Why not put the different steps into smaller helper functions?

Reply:
Sounds like a good idea to me. @HechtiDerLachs do you agree?

Reply:
As discussed on slack, added to issue https://github.com/oscar-system/Oscar.jl/issues/2718 so we can discuss and address this in the future.